src/components/GiftRankingFilter.tsx

```diff
@@ -56,18 +55,30 @@ const TabButton = styled.button<{ selected: boolean }>`
   border: none;
 `;
 
+function getValidStoredValue<T>(
+  key: string,
+  validValues: T[],
+  defaultValue: T
+): T {
+  const stored = localStorage.getItem(key);
+  return validValues.includes(stored as T)
+    ? (stored as T)
+    : defaultValue;
+}
+
 export default function GiftRankingFilter() {
-  const [selected, setSelected] = useState<FilterKey>(() => {
-    const saved = localStorage.getItem(LOCAL_FILTER_KEY);
-    return (
-      filters.some(f => f.key === saved) ? saved : 'all'
-    ) as FilterKey;
-  });
-
-  const [selectedTab, setSelectedTab] = useState(() => {
-    const saved = localStorage.getItem(LOCAL_TAB_KEY);
-    return tabOptions.includes(saved || '') ? saved! : tabOptions[0];
-  });
+  const [selected, setSelected] = useState<FilterKey>(() =>
+    getValidStoredValue(
+      LOCAL_FILTER_KEY,
+      filters.map(f => f.key),
+      'all'
+    )
```

Comment:
바로 아래 tabOptions 코드를 보고 든 생각인데요,
이 부분도 마찬가지로 처리해주시는 것도 좋을 것 같네요!!

```suggestion
    getValidStoredValue(
      LOCAL_FILTER_KEY,
      filters.map(f => f.key),
      filters[0].key
    )
```